vignettes/lock_and_prop.Rmd

```diff
@@ -0,0 +1,74 @@
+
+## Updates to Lock Files 
+
+The intention of the `renv.lock` file is to create a harmonized development environment across our many different companies and collaborators.  The core team feels that if everyone is working with the same version of R and the same versions of R packages then communication around development issues will be much easier.  Unfortunately, this adoption of a `renv.lock` file also creates a few barriers for our development process.  First, if we **lock** in a version of R and R packages then we can not quickly adopt the latest and greatest of updates from new releases.  Second, how does a team decide to update a certain package or version of R and who gets to make that final call? Finally, when do we make updates to the lock file. Hopefully, the below will provide some guidance around these issues.  However, these process might change over time and admiral developers are encouraged to stay up to date with the latest processes and participate in the discussion.
+
+**Note:** Through GitHub Actions we continuously test admiral's integration with the latest 3 snapshots of dependent R packages on CRAN that are closest in date to latest 3 versions of R.  Please see [GitHub Actions/Workflows]( https://pharmaverse.github.io/admiraldev/devel/articles/pr_review_guidance.html#github-actionsworkflows) on the `{admiraldev}` site for more information on this process.
+
+### R Package Updates and Lock Files
+
+As you gathered, package updates to the `renv.lock` file are permissible and needed as our family grows and dependent packages and R versions change.  
+
+Most of the scenarios for updates can be slotted into the following:
+
+1. A new function in a dependent package, e.g. `{dplyr}` is available and could be utilized in a new admiral function.
+1. A new package is available with new functions that could greatly simplify our code base or add in additional functionality.
+1. A breaking bug is discovered in a dependent package that breaks backward compatibility for that package with admiral.
+1. Other?
+
+The first step for requesting an update for a R package in the `renv.lock` file is to raise it in our [slack channel](https://app.slack.com/client/T028PB489D3/C028SJ83KM1).  The next step is to make an issue for it in your relevant admiral repository and tag relevant admiral leads.  This will create discussion around why this update is needed. The admiral leads will then make the final decision (**almost always in the affirmative**) and we can move this to the attention of the `{admiralci}` leads.  Once all leads are aware of this needed update, then the lock file can be updated in the respective admiral repository.  Other admiral family repositories can choose to also include this update in their lock files.  However, if this update is not made then the propagation process will automatically update it for all admiral family packages at a later time (discussed below).
```

Comment:
Why should the lock file be updated in the _respective admiral repository_? Wouldn't this cause issues? If we update the lock file in the respective admiral repository and a propagation is started by a different developer, you could get merge conflicts or the generated PR from the propagation process could revert the changes in the respective admiral repository.
I wonder if the issue should be created in admiralci together with a PR of the proposed changes.

Reply:
I was thinking that an update could be made to a specifc repo lock file for say `admiravaccines` as they want function x from a more updated `tidyr` for a new function that team is developing.  As `admiral` doesn't need this `tidyr` function we don't need to update our lock file.  Once we do a release then everything could get updated and "re-centralized" and all admiral repos have same lock file to start with a new release.   This way a team is not slowed down in their development.

But...

I am partial to a more centrally-controlled process that you are suggesting.  If an update is requested, then everyone gets that update no matter what.  Honestly, this makes more sense and might be way easier to maintain.  

So...update requested for a package, discussion initiated in an admiralci issue with relevant parties tagged, agree on update,  propagation process is initiated across all admiral repos.  Rinse and repeat during development cycle

I think when we do a release we should grab newer package updates so we have the latest and greatest material.